src/rtc/mosaic_geobursts.py

```diff
@@ -0,0 +1,321 @@
+'''
+A module to mosaic Sentinel-1 geobursts from RTC workflow
+'''
+
+import os
+
+import numpy as np
+from osgeo import osr, gdal
+
+def check_reprojection(list_rtc_imagery: list, list_nlooks: list,
+                             geogrid_mosaic) -> bool:
+    '''
+    Check if the reprojection is necessary to mosaic the input list of the rasters
+
+    Parameters:
+    -----------
+    list_rtc_imagery: list
+        path to the geocoded RTC images to mosaic
+    list_nlooks: list
+        path to the nlooks raster that corresponds to list_rtc
+
+    Returns:
+    flag_rtn: bool
+        True if reprojection is necessary to mosaic `list_rtc_imagery`;
+        False if the images are aligned, so that no reprojection is necessary.
+    '''
+
+    # Accepted error in the coordinates as floating number. Used when checking the snapping.
+    maxerr_coord = 1.0e-6
+
+    # Return value after this mosaicking eligibility check.
+    # It will turn to False if any of the checks below fails.
+    flag_rtn = False
```

Comment:
It's better to choose a meaningful variable name, such as `flag_requires_reprojection` or something similar

Reply:
I've changed the name of the return value. Also the variables `*_prev` has been changed to `*_0` because they are not actually "previous" values, but the values from the 1st image.

Also some explanation in the comments are revised for better clarity.